subsys/bluetooth/audio/csis.c

```diff
@@ -557,14 +557,14 @@ static void disconnect_adv(struct k_work *work)
 static void handle_csis_disconnect(struct bt_csis *csis, struct bt_conn *conn)
 {
 #if defined(CONFIG_BT_EXT_ADV)
-	__ASSERT(csis->srv.conn_cnt != 0, "Invalid csis->srv.conn_cnt value");
+	__ASSERT(csis->srv.conn_cnt > 0, "conn_cnt is 0");
 
-	if (csis->srv.conn_cnt == CONFIG_BT_MAX_CONN &&
-	    csis->srv.adv_enabled) {
```

Comment:
I think the removal of the `csis->srv.conn_cnt == CONFIG_BT_MAX_CONN` check here does so that `k_work_submit(&csis->srv.work);` is called multiple times, which would give errors.

Reply:
When k_work_submit(&csis->srv.work); shall be called? I thought that is should be called when device can accept yet another connection.

Reply:
@Thalley Could you please elaborate why you expect errors?
I've set `CONFIG_BT_MAX_CONN=3` for testing purposes and gave it a try against Android. I did not see errors.

But once connected I do not see the device advertises. Is that as expected?

Reply:
submitting `srv.work` will call `disconnect_adv`, which will in turn call `csis_adv_resume`. 

If we are already advertising, `csis_adv_resume` will return an error (cannot enable advertising when it is already enabled), and we should see this error:

```c
	err = csis_adv_resume(csis);

	if (err != 0) {
		BT_ERR("Disconnect: Could not restart advertising: %d",
			err);
		csis->srv.adv_enabled = false;
	}
```

So the `srv.work` work should only be submitted if, and only if we stopped advertising when `conn_cnt == CONFIG_BT_MAX_CONN`, i.e. we should only call `csis_adv_resume` if `conn_cnt == CONFIG_BT_MAX_CONN` when the disconnect occurs

Reply:
I left this check as it was before